pennylane/estimator/ops/qubit/parametric_ops_single_qubit.py

```diff
@@ -0,0 +1,760 @@
+
+        .. math:: R_\phi(\phi) = e^{i\phi/2}R_z(\phi) = \begin{bmatrix}
+                    1 & 0 \\
+                    0 & e^{i\phi}
+                \end{bmatrix}.
+
+    .. seealso:: The corresponding PennyLane operation :class:`~.pennylane.PhaseShift`.
+
+    **Example**
+
+    The resources for this operation are computed as:
+
+    >>> qml.estimator.PhaseShift.resource_decomp()
+    [(1 x RZ), (1 x GlobalPhase)]
+    """
+
+    num_wires = 1
+    resource_keys = {"precision"}
+
+    def __init__(self, precision: float | None = None, wires: WiresLike = None) -> None:
+        self.precision = precision
+        super().__init__(wires=wires)
+
+    @property
+    def resource_params(self) -> dict:
+        r"""Returns a dictionary containing the minimal information needed to compute the resources.
+
+        Returns:
+            dict: A dictionary containing the resource parameters:
+                * precision (Union[float, None]): the number of qubits the operation is controlled on
+        """
+        return {"precision": self.precision}
+
+    @classmethod
+    def resource_rep(cls, precision=None) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the operator that are needed to compute the resources.
+
+        Args:
+            precision (float, optional): The error threshold for the Clifford + T decomposition of this operation.
+
+        Returns:
+            :class:`~.pennylane.estimator.resource_operator.CompressedResourceOp`: A compressed representation of the operator.
+        """
+
+        return CompressedResourceOp(cls, cls.num_wires, {"precision": precision})
+
+    @classmethod
+    def resource_decomp(cls, precision=None) -> list[GateCount]:
+        r"""Returns a list representing the resources of the operator. Each object represents a quantum gate
+        and the number of times it occurs in the decomposition.
+
+        Keyword Args:
+            precision (float): error threshold for the Clifford + T decomposition of this operation
+
+        Resources:
+            The phase shift gate is equivalent to a Z-rotation upto some global phase,
+            as defined in the following identity:
+
+            .. math:: R_\phi(\phi) = e^{i\phi/2}R_z(\phi) = \begin{bmatrix}
+                        1 & 0 \\
+                        0 & e^{i\phi}
+                    \end{bmatrix}.
+
+        Returns:
+            list[:class:`~.pennylane.estimator.resource_operator.GateCount`]: A list of ``GateCount`` objects,
+            where each object represents a specific quantum gate and the number of times it appears
+            in the decomposition.
+        """
+        rz = resource_rep(RZ, {"precision": precision})
+        global_phase = resource_rep(GlobalPhase)
+        return [GateCount(rz), GateCount(global_phase)]
+
+    @classmethod
+    def adjoint_resource_decomp(cls, target_resource_params) -> list[GateCount]:
```

Comment:
Same here, mind updating this?
```suggestion
    def adjoint_resource_decomp(cls, target_resource_params: dict) -> list[GateCount]:
```